plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -316,6 +319,41 @@ public boolean accept(File dir, String name) {
     return matchingFiles;
   }
 
+/**
+ * This method is a decorator around the KillAllSpawnedHadoopJobs method.
+ * This method takes additional parameters to determine whether KillAllSpawnedHadoopJobs needs to be executed 
+ * using doAs as a different user
+ * 
+ * @param logFilePath Azkaban log file path
+ * @param jobProps Azkaban job props
+ * @param tokenFile Pass in the tokenFile if value is known.  It is ok to skip if the token file is in the environmental variable
+ * @param log a usable logger
+ */
+  public static void proxyUserKillAllSpawnedHadoopJobs(final String logFilePath, Props jobProps, File tokenFile, final Logger log) {
```

Comment:
Agreed.
Timing wise, I think it should happen when we remove the support for ProcessJob.EXECUTE_AS_USER and ProcessJob.EXECUTE_AS_USER_OVERRIDE